### PR TITLE
3.0.0 Changelog. Adds deprecations and upgrade guide

### PR DESCRIPTION
**Pull request by nhunzaker**
It's here! Pagination! I've written up a changelog and deprecation warnings for old usage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,39 +59,45 @@ ArsArsenal.render(app, {
...
+  listQuery: function({ search, page }) {
+    // Use this function to rename query parameters before building
+    // the listUrl URL
+    //
+    // Any data returned from this function will be stringified into
+    // query parameters
+    return { search, page }
+  },
+
+  showUrl: function(url, slug: ID) {
```
Comment (cwmanning) [inline README.md]: should `ID` match up with `id` below?

File: src/containers/load-collection.tsx [hunk 2]
```diff
@@ -44,13 +44,30 @@ class CollectionFetcher extends React.Component<Props, State> {
...
+      console.warn(
+        'ArsArsenal option makeURL is deprecated. Use listUrl instead.'
+      )
+    }
+
+    if ('makeQuery' in props) {
+      queryString = props.makeQuery(search)
+
+      console.warn(
+        'ArsArsenal option makeQuery is deprecated. Use buildQuery instead.'
```
Comment (cwmanning) [inline src/containers/load-collection.tsx]: 👏

**Comment by nhunzaker** [comment]
@cwmanning great idea!

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,85 @@
...
+sorting would cause a frustrating reordering of items as new data
+loads.
+
+To enable sorting, take advantage of the `sort` field in the
+`buildQuery` method:
+
+```javascript
+function buildQuery({ page, search, sort }) {
+  return { page, search, sort }
+}
```
Comment (dbernheisel) [inline CHANGELOG.md]: Is there a fuller example we can provide?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,85 @@
...
+let listUrl = url => url
+let showUrl = (url, slug) => `${url}/${slug}`
+
+ArsArsenal.render({ listUrl, showUrl })
+```
+
+**These are the default implementations of each option.** We anticipate
+that this change affects very few users.
+
+### `makeQuery` is now `buildQuery` and returns an object
```
Comment (cwmanning) [inline CHANGELOG.md]: I believe this is now `listQuery`

Comment (cwmanning) [reply CHANGELOG.md]: and if that's the case, there are a couple of additional references to do a quick search/replace on

File: README.md (hunk 1 above)
Comment (nhunzaker) [reply README.md]: Great idea. Done.

File: CHANGELOG.md (hunk 3 above)
Comment (nhunzaker) [reply CHANGELOG.md]: Added a more descriptive example and also included it in the readme.

**Comment by nhunzaker** [comment]
Thanks everyone! I'll send out a stable release shortly.

**Pull request merged**